dask/array/creation.py

```diff
@@ -101,6 +101,12 @@ def empty_like(a, dtype=None, order="C", chunks=None, name=None, shape=None):
 
     a = asarray(a, name=False)
     shape, chunks = _get_like_function_shapes_chunks(a, chunks, shape)
+
+    # if shape is nan we cannot rely on regular empty function, we use
+    # generic map_blocks.
+    if np.isnan(shape).any():
+        return a.map_blocks(lambda x: np.empty_like(x, dtype=(dtype or x.dtype)))
```

Comment:
Can we avoid the `lambda` use here? This will cause the name of the output tasks to be `lambda` instead of `empty_like`. It's also more expensive to pickle `lambda`s, but the collection name is my main concern. Can we just pass `np.empty_like` to `map_blocks`? 

I have a similar comment for the other `*_like` methods too

Reply:
Ah yes this makes a lot of sense. I've tinkered with this but the one issue is `dtype=` is an explicit argument absorbed by `map_blocks` itself, so it doesn't get passed down as an argument to the callable as other `**kwargs` do. Should we `partial(np.foo_like, dtype=...)` the callables? Is that better than a lambda?